contrib/NeonVM/neonvm.c

```diff
@@ -255,27 +261,87 @@ int main(int argc, char* argv[])
 void exec_run(struct tagTExecutor *self)
 {
     uint32_t i;
-    push(self->callstack, &self->object->codelen);
+    self->callstack[self->callstacktop] = (uint32_t)self->object->codelen;
+
     exec_loop(self);
 
     for (i = 0; i < self->object->strtablelen; i++) {
         free(self->object->strtable[i]);
     }
+    for (i = 0; i < self->object->global_size; i++) {
+        free(self->globals[i]);
+    }
 }
 
 struct tagTExecutor *new_executer(struct tagTBytecode *object)
 {
+    uint32_t i;
     struct tagTExecutor *r = malloc(sizeof(struct tagTExecutor));
     if (r == NULL) {
         fatal_error("Failed to allocate memory for Neon executable.");
     }
     r->object = object;
-    r->callstack = createStack(300);
     r->stack = createStack(300);
     r->ip = 0;
+    r->callstacktop = 0;
+    r->globals = malloc(sizeof(Cell) * r->object->global_size);

```

Comment:
Looks like this should be sizeof(Cell *) instead of sizeof(Cell), since globals in array of pointers.